test/test_cuda.py

```diff
@@ -738,6 +738,50 @@ def test_caching_pinned_memory_multi_gpu(self):
         self.assertEqual(gpu_tensor1[0], 1)
         self.assertEqual(gpu_tensor0[0], 2)
 
+    def test_btrifact(self):
+        a = torch.Tensor((((1.3722, -0.9020),
+                           (1.8849, 1.9169)),
+                          ((0.7187, -1.1695),
+                           (-0.0139, 1.3572)),
+                          ((-1.6181, 0.7148),
+                           (1.3728, 0.1319)))).cuda()
+        LU_data, pivots, info = a.btrifact()
+        self.assertEqual(info.abs().sum(), 0)
+        I_U = torch.triu(torch.ones(2, 2)).unsqueeze(0).expand(3, 2, 2).type_as(a).byte()
+        I_L = 1 - I_U
+        a_L = torch.zeros(a.size()).type_as(a)
+        a_U = a_L.clone()
+        a_L[torch.eye(2).unsqueeze(0).expand(3, 2, 2).type_as(a).byte()] = 1.0
+        a_L[I_L] = LU_data[I_L]
+        a_U[I_U] = LU_data[I_U]
+
+        P = torch.eye(2).unsqueeze(0).expand(3, 2, 2).type_as(a)
+        for i in range(3):
+            for j in range(2):
+                k = pivots[i, j] - 1
```

Comment:
Yes, this is standard in LAPACK's C interface (http://www.netlib.org/clapack/old/double/dgetrf.c) and cublas (http://docs.nvidia.com/cuda/cublas/#cublas-lt-t-gt-getrfbatched)